resource-aggregate/events/resourceStateSnapshotTaken.go

```diff
@@ -536,78 +532,123 @@ func (e *ResourceStateSnapshotTaken) ConfirmResourceCreateCommand(ctx context.Co
 			events = append(events, ev...)
 		}
 	}
+	return events
+}
+
+func (e *ResourceStateSnapshotTaken) CancelResourceDeleteCommand(ctx context.Context, events []eventstore.Event, userID string, correlationIdFilter strings.Set, resourceID *commands.ResourceId, commandMetadata *commands.CommandMetadata, status commands.Status, content *commands.Content, newVersion uint64) []eventstore.Event {
 	for _, event := range e.GetResourceDeletePendings() {
 		if len(correlationIdFilter) != 0 && !correlationIdFilter.HasOneOf(event.GetAuditContext().GetCorrelationId()) {
 			continue
 		}
 		ev, err := e.ConfirmResourceDeleteCommand(ctx, userID, &commands.ConfirmResourceDeleteRequest{
-			ResourceId:      req.GetResourceId(),
+			ResourceId:      resourceID,
 			CorrelationId:   event.GetAuditContext().GetCorrelationId(),
-			Status:          commands.Status_CANCELED,
-			CommandMetadata: req.GetCommandMetadata(),
+			Status:          status,
+			CommandMetadata: commandMetadata,
+			Content:         content,
 		}, newVersion+uint64(len(events)))
 		if err == nil {
 			// errors appears only when command with correlationID doesn't exist
 			events = append(events, ev...)
 		}
 	}
+	return events
+}
+
+func (e *ResourceStateSnapshotTaken) CancelPendingCommands(ctx context.Context, userID string, req *commands.CancelPendingCommandsRequest, newVersion uint64) ([]eventstore.Event, error) {
+	if req.GetCommandMetadata() == nil {
+		return nil, status.Errorf(codes.InvalidArgument, errInvalidCommandMetadata)
+	}
+	correlationIdFilter := strings.MakeSet(req.GetCorrelationIdFilter()...)
+	events := make([]eventstore.Event, 0, 4)
+	events = e.CancelResourceCreateCommand(ctx, events, userID, correlationIdFilter, req.GetResourceId(), req.GetCommandMetadata(), newVersion)
+	events = e.CancelResourceUpdateCommand(ctx, events, userID, correlationIdFilter, req.GetResourceId(), req.GetCommandMetadata(), newVersion)
+	events = e.CancelResourceRetrieveCommand(ctx, events, userID, correlationIdFilter, req.GetResourceId(), req.GetCommandMetadata(), newVersion)
+	events = e.CancelResourceDeleteCommand(ctx, events, userID, correlationIdFilter, req.GetResourceId(), req.GetCommandMetadata(), commands.Status_CANCELED, nil, newVersion)
 	if len(events) == 0 {
 		return nil, status.Errorf(codes.NotFound, "cannot find commands with correlationID(%v)", req.GetCorrelationIdFilter())
 	}
 	return events, nil
 }
 
+func (e *ResourceStateSnapshotTaken) isEmpty() bool {
+	return e.GetLatestResourceChange() == nil && len(e.GetResourceCreatePendings()) == 0 && len(e.GetResourceDeletePendings()) == 0 && len(e.GetResourceRetrievePendings()) == 0 && len(e.GetResourceUpdatePendings()) == 0
+}
+
+func (e *ResourceStateSnapshotTaken) UnpublishResource(ctx context.Context, userID string, resourceID *commands.ResourceId, commandMetadata *commands.CommandMetadata, deletedContent *commands.Content, newVersion uint64) ([]eventstore.Event, error) {
```

Comment:
This function is not unpublishing resources. Please name it by describing what it does, not when it is used.

Reply:
fixed